rest-api-spec/src/main/resources/rest-api-spec/api/index.json

```diff
@@ -10,6 +10,7 @@
         {
           "path":"/{index}/_doc/{id}",
           "methods":[
+            "POST",
```

Comment:
The rule here was and should remain the first `HTTP method` in the array should be the preferred one, so i think `PUT` should come before `POST` here.

Reply:
aren't these methods left in the same order as they were before the change?

Reply:
`index.json` always needed special handling in our generators because methods was for the whole API not per endpoint. 

Being able to put the preferred HTTP method for `{index}/_doc/{id}` is one of the benefits of the new format :smiley:

Reply:
ok lets keep this as a follow-up please. I just got a green build on this huge change after merging, re-merging, running and re-running tests :)

Reply:
Cool cool, will submit it after we get this merged :+1:

Reply:
thanks!